crowbar_framework/app/models/api/node.rb

```diff
@@ -142,11 +142,44 @@ def remove_pre_upgrade_role_for(name)
     # Delete existing pacemaker resources, from other node in the cluster
     def delete_pacemaker_resources
       script = "/usr/sbin/crowbar-delete-pacemaker-resources.sh"
-      out = @node.run_ssh_cmd(script)
-      unless out[:exit_code].zero?
-        save_error_state("Deleting pacemaker resources failed on node #{@node.name}.")
-        Rails.logger.error("Script location: #{script}")
-        Rails.logger.error("stdout: #{out[:stdout]}, stderr: #{out[:stderr]}")
+      ssh_status = @node.ssh_cmd(script)
+      if ssh_status[0] != 200
+        save_error_state(
+          "Executing of 'delete pacemaker resources' has failed at node #{@node.name}.\
+Script location: #{script}."
+        )
+        return false
+      end
+
+      # wait until the the script is finished
+      upgrade_failure = false
+      begin
+        Timeout.timeout(300) do
+          loop do
+            out = @node.run_ssh_cmd(
+              "test -e /var/lib/crowbar/upgrade/pacemaker-resources-delete-ok"
+            )
+            if out[:exit_code].zero?
+              save_node_state("Deleting pacemaker resources was successful.")
+              break
+            end
+            out = @node.run_ssh_cmd(
+              "test -e /var/lib/crowbar/upgrade/pacemaker-resources-delete-failed"
```

Comment:
Code duplication.  Would be cleaner to extend the model and then write `@node.file_exists("/var/lib/crowbar/...")`.